platform/src/register.rs

```diff
@@ -0,0 +1,62 @@
+/// In order to work with Miri's `-Zmiri-track-raw-pointers` flag, we cannot
+/// pass pointers to the kernel through `usize` values (as casting to and from
+/// `usize` drops the pointer`s tag). Instead, `RawSyscalls`
+/// uses the `Register` type. `Register` is an alias for a raw pointer type that
+/// keeps that tags around. User code should not depend on the particular type
+/// of pointer that `Register` aliases, but instead use the conversion functions
+/// in this module.
+pub type Register = *mut ();
```

Comment:
Ultimately, the reason is "For the second time, I forgot about destructuring in the `RawSyscalls` implementation".

That said, using a newtype wrapper makes the `RawSyscalls` implementation [more complex](https://github.com/tock/libtock-rs/pull/290/files#diff-9d498edee15500dae0e43113da1336bdcf5235d742518cf9da77f24ca370504f). I'm weary of complexity in `RawSyscalls` because it is untestable and very `unsafe`.

Therefore, I made that change in a new PR (#290), and we can collectively decide which PR we want to merge.